File: app/src/main/java/com/example/adarsh/rajamantrichorsipahi/MainActivity.java
package com.example.adarsh.rajamantrichorsipahi;

import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;


public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        View decorView = getWindow().getDecorView();
// Hide both the navigation bar and the status bar.
// SYSTEM_UI_FLAG_FULLSCREEN is only available on Android 4.1 and higher, but as
// a general rule, you should design your app to hide the status bar whenever you
// hide the navigation bar.
        int uiOptions = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_FULLSCREEN | View.SYSTEM_UI_FLAG_IMMERSIVE;
        decorView.setSystemUiVisibility(uiOptions);
        final EditText ed1 = (EditText) findViewById(R.id.edittext1);
        final EditText ed2 = (EditText) findViewById(R.id.edittext2);
        final EditText ed3 = (EditText) findViewById(R.id.edittext3);
        final EditText ed4 = (EditText) findViewById(R.id.edittext4);
        final TextView tv = (TextView) findViewById(R.id.tv);
        Button tx = (Button) findViewById(R.id.button);
        Typeface custom_font = Typeface.createFromAsset(getAssets(), "fonts/xenippa1.ttf");
        tx.setTypeface(custom_font);
        ed1.setTypeface(custom_font);
        ed2.setTypeface(custom_font);
        ed3.setTypeface(custom_font);
        ed4.setTypeface(custom_font);
        tv.setTypeface(custom_font);
        ed1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ed1.setText("");
                ed1.setCursorVisible(true);
            }
        });
        ed2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ed2.setText("");
                ed2.setCursorVisible(true);
            }
        });
        ed3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ed3.setText("");
                ed3.setCursorVisible(true);
            }
        });
        ed4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ed4.setText("");
                ed4.setCursorVisible(true);
            }
        });
        tx.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (ed1.getText().length() == 0)
                    Toast.makeText(getApplicationContext(), "Enter Player 1 Name", Toast.LENGTH_SHORT).show();
                else if (ed2.getText().length() == 0)
                    Toast.makeText(getApplicationContext(), "Enter Player 2 Name", Toast.LENGTH_SHORT).show();
                else if (ed3.getText().length() == 0)
                    Toast.makeText(getApplicationContext(), "Enter Player 3 Name", Toast.LENGTH_SHORT).show();
                else if (ed4.getText().length() == 0)
                    Toast.makeText(getApplicationContext(), "Enter Player 4 Name", Toast.LENGTH_SHORT).show();
                else {

                    Intent intent = new Intent(MainActivity.this, guess.class);
                    intent.putExtra("edittext1", ed1.getText().toString());
                    intent.putExtra("edittext2", ed2.getText().toString());
                    intent.putExtra("edittext3", ed3.getText().toString());
                    intent.putExtra("edittext4", ed4.getText().toString());
                    startActivity(intent);
                }
            }
        });
    }
}

File: app/src/main/java/com/example/adarsh/rajamantrichorsipahi/MainActivity.kt
package com.example.adarsh.rajamantrichorsipahi

import android.content.Intent
import android.graphics.Typeface
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var firstPerson: EditText
    private lateinit var secondPerson: EditText
    private lateinit var thirdPerson: EditText
    private lateinit var fourthPerson: EditText
    private lateinit var enterPlayerNameTV: TextView
    private lateinit var submitButton: Button
    private lateinit var decorView: View
    private lateinit var customFont: Typeface
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initializeViews();
    }

    private fun initializeViews() {
        firstPerson = findViewById(R.id.firstPerson)
        secondPerson = findViewById(R.id.secondPerson)
        thirdPerson = findViewById(R.id.thirdPerson)
        fourthPerson = findViewById(R.id.fourthPerson)
        enterPlayerNameTV = findViewById(R.id.enterPlayerName)
        submitButton = findViewById(R.id.submitButton)
        decorView = window.decorView
        val uiOptions = (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_FULLSCREEN or View.SYSTEM_UI_FLAG_IMMERSIVE)
        decorView.systemUiVisibility = uiOptions
        customFont = Typeface.createFromAsset(assets, "fonts/xenippa1.ttf")
        setCustomFont(customFont)
        firstPerson.setOnClickListener(this)
        secondPerson.setOnClickListener(this)
        thirdPerson.setOnClickListener(this)
        fourthPerson.setOnClickListener(this)
        submitButton.setOnClickListener(this)
    }

    private fun setCustomFont(customFont: Typeface?) {
        firstPerson.typeface = customFont
        secondPerson.typeface = customFont
        thirdPerson.typeface = customFont
        fourthPerson.typeface = customFont
        enterPlayerNameTV.typeface = customFont
        submitButton.typeface = customFont
    }

    override fun onClick(view: View?) {
        when (view?.id) {
            R.id.firstPerson -> {
                firstPerson.setText("")
                firstPerson.isCursorVisible = true
            }
            R.id.secondPerson -> {
                secondPerson.setText("")
                secondPerson.isCursorVisible = true
            }
            R.id.thirdPerson -> {
                thirdPerson.setText("")
                thirdPerson.isCursorVisible = true
            }
            R.id.fourthPerson -> {
                fourthPerson.setText("")
                fourthPerson.isCursorVisible = true
            }
            R.id.submitButton -> {
                when {
                    firstPerson.text.isEmpty() -> {
                        showToast("Enter Player 1 Name")
                    }
                    secondPerson.text.isEmpty() -> {
                        showToast("Enter Player 2 Name")
                    }
                    thirdPerson.text.isEmpty() -> {
                        showToast("Enter Player 3 Name")
                    }
                    fourthPerson.text.isEmpty() -> {
                        showToast("Enter Player 4 Name")
                    }
                    else -> {
                        val intent = Intent(this@MainActivity, GameActivity::class.java).apply {
                            putExtra("firstPerson", firstPerson.text.toString())
                            putExtra("secondPerson", secondPerson.text.toString())
                            putExtra("thirdPerson", thirdPerson.text.toString())
                            putExtra("fourthPerson", fourthPerson.text.toString())
                        }
                        startActivity(intent)
                    }
                }
            }
        }
    }

    private fun showToast(toastMessage: String) {
        Toast.makeText(applicationContext, toastMessage, Toast.LENGTH_SHORT).show()
    }
}